larcore/Geometry/GeometryConfigurationWriter_service.cc

```diff
@@ -0,0 +1,254 @@
+#include "art/Framework/Principal/Handle.h"
+#include "canvas/Utilities/InputTag.h"
+#include "messagefacility/MessageLogger/MessageLogger.h"
+
+// C/C++ standard libraries
+#include <memory> // std::make_unique()
+
+
+// -----------------------------------------------------------------------------
+namespace geo { class GeometryConfigurationWriter; }
+/**
+ * @brief Writes geometry configuration information into _art_ runs.
+ * 
+ * This service is part of the mandatory version check of `geo::Geometry`
+ * service.
+ * It does not require any special configuration, but it must be listed
+ * in the configuration in order for `Geometry` to work:
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ * services: {
+ *   
+ *   GeometryConfigurationWriter: {}
+ *   
+ *   Geometry: @local::experiment_geometry
+ *   
+ *   # ...
+ *  
+ * }
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ * 
+ * The configuration check is described in the documentation of `geo::Geometry`
+ * service.
+ * 
+ * 
+ * Produced data products
+ * -----------------------
+ * 
+ * The service guarantees that configuration information of type
+ * `sumdata::GeometryConfigurationInfo` is present into the run, accessible with
+ * an input tag `GeometryConfigurationWriter`:
+ * 
+ * * if such information is already available in the run, no further information
+ *   is added
+ * * legacy: if there is no such information, but there is a `sumdata::RunData`
+ *   data product, a reduced version of the configuration information is created
+ *   from the information in that data product (the first one, if multiple are
+ *   present)
+ * * finally, if no information is present neither in the full 
+ *   `sumdata::GeometryConfigurationInfo` form nor in the legacy
+ *   `sumdata::RunData` form, information is put together based on the current
+ *   configuration of the `Geometry` service.
+ * 
+ * 
+ * Service dependencies
+ * ---------------------
+ * 
+ * * `Geometry` service
+ *   (for obtaining the current configuration to put into the event)
+ * 
+ */
+class geo::GeometryConfigurationWriter: public art::ProducingService {
+  
+    public:
+  
+  /// Service configuration.
+  struct Config {
+    // no configuration parameters at this time
+  };
+  
+  using Parameters = art::ServiceTable<Config>;
+  
+  /// Constructor: gets its configuration and does nothing with it.
+  GeometryConfigurationWriter(Parameters const&);
+  
+  
+    private:
+  
+  /// Alias for the pointer to the data product object to be put into the run.
+  using InfoPtr_t = std::unique_ptr<sumdata::GeometryConfigurationInfo>;
+  
+  
+  // -- BEGIN --- Interface ----------------------------------------------------
+      
+  /// Writes the information from the service configuration into the `run`.
+  virtual void postReadRun(art::Run& run) override;
+  
+  // -- END --- Interface ------------------------------------------------------
```

Comment:
Why is this labeled "Interface"?  This service has no publicly accessible interface.

Reply:
I did not mean interface between class implementation and class caller, but rather between class and base class.
That block was supposed to enclose all base class methods that would be overridden.
Is the term "interface" inappropriate? Feel free to propose a better one.

Reply:
Using the term `override` is already self-documenting--you can only override a base class membee function--thus no additional commentary is necessary.  You could say, "BEGIN --- Overrides ...", but then you're just repeating what's already apparent.

Reply:
Those markers are for fast lookup rather than for documentation.
I have replaced them.